File: Swift/flancast90/collatz.swift
func collatz() -> Array<Int> {

   var nums: [Int] = []

   print("Enter a number: ")
   
   if let collatz_num = readLine() {
   
      if let testcase = Int(collatz_num){
      
        var val = testcase
        while (val != 1) {
            if (val % 2 == 0) {
                val = val/2
                nums.append(val)
            }else {
                val = (3*val)+1
                nums.append(val)
            }
        }
        
      }else {
          print("Please enter an integer")
      }
      
   }
   
   return nums
   
}

print(collatz())

File: Java/flancast90/src/collatz.java
// import scanner for reading user input
import java.util.Scanner; 
// arraylist for arrays of unknown size
import java.util.ArrayList;

// initialise class collatz for taking input and calculating collatz with it.
public class collatz {
	
	static ArrayList<Integer> nums = new ArrayList<>();
	
	public static void main(String[] args) {
		Scanner collatz_num = new Scanner(System.in);  // Create a Scanner object
	    System.out.println("Enter number: ");

	    int test_case = collatz_num.nextInt();  // Read user input
	    while (test_case > 1) {
	    	if (test_case % 2 != 0) {
	    		test_case = (3*test_case)+1;
	    		nums.add(test_case);
	    	}else {
	    		test_case = test_case/2;
	    		nums.add(test_case);
	    	}
	    }
	    System.out.println(nums);
	}
	
}